draft-law-moq-cat4moqt.md

```diff
@@ -289,7 +361,43 @@ TODO Security
 
 # IANA Considerations
 
-TODO We need to register the "moqt" claim in the IANA "CBOR Web Token (CWT) Claim" table.
+IANA will register the following claim in the "CBOR Web Token (CWT) Claims" registry:
+
+|------------------------|----------------|
+|                        | Value          |
+|------------------------|----------------|
+| Claim Name             | moqt           |
+| Claim Description      | MOQT Action    |
+| JWT Claim Name         | N/A            |
+| Claim Key              | TBD_MOQT (1+2) |
+| Claim Value Type       | array          |
+| Change Controller      | IESG           |
+| Specification Document | RFCthis        |
+|------------------------|----------------|
+
+\[RFC Editor: Please replace RFCthis with the published RFC number for this
+document.\]
+
+IANA will create a subregistry within the "\[TODO: Put the name of the registry
+in here when draft-ietf-moq-transport defines it\]" registry. The subregistry
+will be named "MOQT Auth Action Keys". (DISCUSS: If actions are not extensible,
+we can avoid the registry entirely and simply require that new specifications
+that include an action include a key value? Maybe we can straight up reuse the
+Message ID?)
```

Comment:
As mentioned earlier, actions are defined by MOQT RFC and are not extensible. We can require that new specifications include and action and a key value when extending cat-4-moqt. 

Reusing message id is not an option. The message id mutates with each message sent so it would not work for the repeated use of tokens. 
> The client's Request ID starts at 0 and are even and the server's Request ID starts at 1 and are odd. The Request ID increments by 2 with ANNOUNCE, FETCH, SUBSCRIBE, SUBSCRIBE_ANNOUNCES or TRACK_STATUS request.

Reply:
I should have said Message Type IDs here. See Figure 3 in Section 8 of draft-ietf-moq-transport. So, for example, ANNOUNCE is 0x6, FETCH is 0x16, SUBSCRIBE is 0x3, SUBSCRIBE_ANNOUNCES is 0x11, and TRACK_STATUS is 0xE. The advantage is that you don't wind up using different numbers to describe the same concept and it means you don't have to define a list in the auth token. It also makes it effectively automatically extensible if someone defines a new frame in the future.

The main downside is that it allows you to express the idea of having the auth token apply to message types that might not make sense, like "FETCH_ERROR" or "MAX_REQUEST_ID". This could lead to confusion, or in the worst case, a naive implementation causing real problems for themselves.

Reply:
I like idea of reusing the message type, as long as it does not mutate between MOQT versions. There is no danger in having the token apply to message types that don't carry auth precisely because the serialization of those message types doesn't provide for transmission of the auth token.

Reply:
The CONNECT example convinced me that we should just enumerate the correct values here.